etc/tempesta_fw.conf

```diff
@@ -789,20 +808,45 @@
 # TMT is non-negative integer (default value is 0, which means that
 # timeout tracking is disabled).
 #
+# 'options' - string of extra options for Tempesta sticky cookie. Added after
+# cookie value as plain string in 'Set-Cookie' header. The field may contain
+# multiple options, in this case, the options must be split with semi-colon.
+# E.g.: 'options="Domain=<domain-value>; Secure; HttpOnly",
+# 'options="Max-Age=3600"'.
+#
 # NOTE: line breaks inside the directive are not allowed. See 'js_challenge'.
 #
 # Examples:
-#   sticky;                         # Enable sticky cookie with default name.
-#   sticky enforce;                 # Enforce sticky cookie with default name.
-#   sticky name=__cookie__ enforce; # Enable sticky cookie with custom name.
+#   cookie;                         # Enable sticky cookie with default name.
+#   cookie enforce;                 # Enforce sticky cookie with default name.
+#   cookie name=__cookie__ enforce; # Enable sticky cookie with custom name.
+#
+
+# TAG: learn
+#
+# Syntax:
+#   learn name=<COOKIE_NAME>;
+#
+# Learn HTTP sessions created by backend server. When the backend server
+# creates session it sends 'Set-Cookie' header to the client. If the option
+# is enabled, a new HTTP session entry is created at receiving 'Set-Cookie'
+# header and pinned to origin backend server. All requests with the same cookie
+# will be forwarded to the same backend server.
+#
+# The session in Tempesta expires after 'sess_lifetime' seconds after last
```

Comment:
If backend server sends no `Set-Cookie`, then nothing to learn and the request is sent to random server (random means any in the group according to load balancer rules). It's ok, since we don't know the rules, how the session pinning works in backend application. E.g. cookie has a path and the browser will use the cookie only for matching path, i.e. requests to `/my-web-app` will use the cookies and session pinning while `/img/` requests doesn't require session persistence and the fastest performing server can catch up the requests.

Regarding `sess_lifetime` case. we need some limitations upon the session lifetime in TDB. We know nothing about session lifetime on backend servers. We can follow `Expire` cookie parameter, but it can last for ages, and it will consume too lot of TDB space for nothing. E.g.  cookie with format `<client_id>-<pinned-server-id>`, the first part is to be the same for ever, while the second is to be changed on every new browser session. The same limitation Nginx has (option `timeout` for directive `learn` with defaults to 10 minutes).

Reply:
What about learning about existing sessions from cookies sent by clients? As far as I understand "sticky" directive description from Nginx, it has both "create" and "learn" attributes, for discovering sessions newly created by an upstream and for catching up with already existing sessions, respectively.

Unfortunately, sticky sessions support is a part of a commercial subscription in Nginx, no source code is available, so we can't learn how that is implemented in Nginx.